game/scripts/vscripts/abilities/visage_gravekeepers_cloak.lua

```diff
@@ -0,0 +1,109 @@
+
+if Debug then
+  Debug.EnabledModules['abilities:*'] = true
+end
+
+LinkLuaModifier("modifier_visage_gravekeepers_cloak_oaa", "abilities/visage_gravekeepers_cloak.lua", LUA_MODIFIER_MOTION_NONE) --- PETH WEFY INPARFANT
+LinkLuaModifier("modifier_visage_gravekeepers_cloak_oaa_aura", "abilities/visage_gravekeepers_cloak.lua", LUA_MODIFIER_MOTION_NONE) --- PETH WEFY INPARFANT
+
+visage_gravekeepers_cloak_oaa = class(AbilityBaseClass)
+
+function visage_gravekeepers_cloak_oaa:GetIntrinsicModifierName()
+  return "modifier_visage_gravekeepers_cloak_oaa"
+end
+
+if IsServer() then
+  function visage_gravekeepers_cloak_oaa:OnUpgrade()
+    local caster = self:GetCaster()
+    local mod = caster:FindModifierByName("modifier_visage_gravekeepers_cloak_oaa")
+    mod:SetStackCount(self:GetSpecialValueFor("max_layers"))
+  end
+end
+
+---------------------------------------------------------------
+
+modifier_visage_gravekeepers_cloak_oaa = class(ModifierBaseClass)
+
+function modifier_visage_gravekeepers_cloak_oaa:OnCreated()
+  self:SetStackCount(self:GetAbility():GetSpecialValueFor("max_layers"))
+end
+
+modifier_visage_gravekeepers_cloak_oaa.OnRefresh = modifier_visage_gravekeepers_cloak_oaa.OnCreated
+
+function modifier_visage_gravekeepers_cloak_oaa:DeclareFunctions()
+  return {
+    MODIFIER_PROPERTY_TOTAL_CONSTANT_BLOCK
+  }
+end
+
+function modifier_visage_gravekeepers_cloak_oaa:GetModifierTotal_ConstantBlock(keys)
+  local ability = self:GetAbility()
+  local caster = self:GetCaster()
+  local stackCount = self:GetStackCount()
+  local damageReduction = math.min(100, self:GetAbility():GetSpecialValueFor("damage_reduction") * stackCount)
+  local damageThreshold = self:GetAbility():GetSpecialValueFor("minimum_damage")
+  if keys.attacker:GetTeam() ~= caster:GetTeam() and (keys.attacker:GetTeam() == DOTA_TEAM_GOODGUYS or keys.attacker:GetTeam() == DOTA_TEAM_BADGUYS) then
```

Comment:
I think there are ways to properly check if the attacker is player-owned, but this works too I suppose, though it doesn't technically check for player-owned.